src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Unix.cs

```diff
@@ -358,6 +359,43 @@ internal static unsafe InnerSafeCloseSocket Accept(SafeSocketHandle socketHandle
                 res.SetHandle(acceptedFd);
                 return res;
             }
+
+            internal unsafe void TryUnblockSocket(bool abortive)
+            {
+                // Calling 'close' on a socket that has pending blocking calls (e.g. recv, send, accept, ...)
+                // may block indefinitly. This is a best effort attempt to not get blocked and make those operations return.
+                // We need to ensure we keep the expected TCP behavior that is observed by the socket peer (FIN vs RST close).
+                // What we do here isn't specified by POSIX, it works for Linux.
```

Comment:
> it works for Linux

What happens on macOS?  On FreeBSD?  Does it help?  Hurt?  Nop?

Reply:
It shouldn't hurt.